catalog/core/catalog-core-api/src/main/java/ddf/catalog/service/TransformerLocator.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.service;
```

Comment:
❗️Since the `TransformerLocator` is not meant to be used by the catalog framework clients, but is rather meant to be an internal service, I would recommend putting the interface and its implementation and related classes in different modules, e.g., `catalog-core-transformer-locator-api` and `catalog-core-transformer-locator-impl`. That will allow us to evolve it over time without impacting the catalog API, and will also prevent other problems such as component tests trying to mock out the catalog framework and having to also mock the `TransformerLocator` interface even though they don't need or use it.

Reply:
@coyotesqrl @pklinef How do you feel about Eric's suggestion to move the new TransformerLocator into separate bundles?

Reply:
Sounds good to me.

Reply:
I agree.  I think you should create a new `catalog-core-transformer-api` that uses a `org.codice.ddf.catalog.transform` package.  Eventually we would migrate the transformer interfaces from `ddf.catalog.transform` in the catalog core API to this new transformer API bundle.